### PR TITLE
add documentation strings to classes and methods in check_stability.py

### PR DESCRIPTION
**Pull request by bobholt**
Documentation-only PR.

My Python's a bit rusty, so I figured I'd walk through check_stability line-by-line in execution order to get familiar with it and knock off a little rust. I figured while I was figuring out what everything did, I might as well add and update docstrings for all classes and methods.

### CODE REVIEW

**Review by jugglinmike — COMMENTED** [review]
Great idea, @bobholt! Thanks for doing this :)

**Comment by bobholt** [comment]
Thanks @jugglinmike!

File: check_stability.py [hunk 1]
```diff
@@ -339,7 +341,7 @@ def call(*args):
 
 
 def get_git_cmd(repo_path):
-    """Create an interface for invoking git commands as a subprocess."""
+    """Create an function for invoking git commands as a subprocess."""
```
Comment (jugglinmike) [inline check_stability.py]: Typo: "an function" -> "a function"

**Comment by jugglinmike** [comment]
Certainly

**Comment by bobholt** [comment]
Augh! (fixed)

**Comment by sideshowbarker** [comment]
@bobholt can merge this but it first needs be rebased against master

File: check_stability.py [hunk 2]
```diff
@@ -58,38 +60,57 @@ def setup_logging():
...
 
 def setup_action_filter():
+    """Create global LogActionFilter class as part of deferred module load."""
     global LogActionFilter
 
     class LogActionFilter(BaseHandler):
-        """Handler that filters out messages with action of log and a level
-        lower than some specified level.
+
+        """Filter out messages not of a given set of actions.
```
Comment (sideshowbarker) [inline check_stability.py]: > It's probably worth checking with @jgraham on this, but the prevailing pattern in this file seems to be: "describe classes as nouns and functions as verbs."

Yeah

**Comment by bobholt** [comment]
@sideshowbarker Rebased.

**Pull request merged**